code/modules/urist/modules/shipbattles/overmapships/components.dm

```diff
@@ -160,4 +167,242 @@
 /datum/shipcomponents/point_defence/alienheavy
 	name = "advanced alien point defence systems"
 	intercept_chance = 25
-	health = 250
\ No newline at end of file
+	health = 250
+
+//boarding
+/datum/shipcomponents/teleporter/robotic
+	name = "robotic boarding module"
+	boarding_mobs = list(/mob/living/simple_animal/hostile/hivebot, /mob/living/simple_animal/hostile/hivebot/range, /mob/living/simple_animal/hostile/hivebot/strong, /mob/living/simple_animal/hostile/hivebot/tele)
+	boarding_number = 8
+
+/datum/shipcomponents/teleporter/alien
+	name = "alien matter deconstructor/reconstructor"
+	boarding_mobs = list(/mob/living/simple_animal/hostile/scom/lactera/light, /mob/living/simple_animal/hostile/scom/lactera/medium, /mob/living/simple_animal/hostile/scom/lactera/heavy)
+	boarding_number= 5
+	boarding_delay = 2.5 MINUTES
+
+/datum/shipcomponents/teleporter/pirate
+	name = "pirate teleporter"
+	boarding_mobs = list(/mob/living/simple_animal/hostile/pirate, /mob/living/simple_animal/hostile/pirate/ranged)
+	boarding_number = 6
+	boarding_delay = 1.5 MINUTES
+
+/datum/shipcomponents/teleporter
+	name = "teleporter module"
+	var/boarding_delay = 3 MINUTES //How long between boarding attempts?
+	var/list/boarding_mobs = list(/mob/living/simple_animal/hostile/russian, /mob/living/simple_animal/hostile/russian/ranged) //What mobs do we spawn on boarding?
+	var/boarding_number = 4 //We'll spawn this many every time we board.
+	var/last_boarding = 0 //How long since we boarded last?
+	var/boarding_turf = null//Where do we spawn our mobs?
+	var/boarding_failure_chance = 0 //more shields = more chance of preventing teleportation.
+
+/datum/shipcomponents/teleporter/DoActivate()
+	if(last_boarding > world.time) //Are we too early?
+		return
+
```

Comment:
I haven't read through all of this yet, but I think you should add a check here for if the mastership is being boarded already

something like `if(mastership?.boarded) return`